pkg/apis/triggers/v1alpha1/event_listener_defaults.go

```diff
@@ -30,6 +30,7 @@ func (el *EventListener) SetDefaults(ctx context.Context) {
 		for i := range el.Spec.Triggers {
 			triggerSpecBindingArray(el.Spec.Triggers[i].Bindings).
 				defaultBindings()
+			templateNameToRef(el.Spec.Triggers[i].Template)
```

Comment:
Call this function only if there's Template isn't nil. I think that's why test is failing.

Reply:
Yes its failing because of that

Updating